mlir/utils/jenkins/MIOpenDriver.py

```diff
@@ -302,45 +303,75 @@ def benchmarkMIOpen(commandLine, xdlops):
-    plotMean = df[COLUMNS_TO_AVERAGE].agg(reportUtils.geoMean)
-    plotMean.name = "Geo. mean"
-    plotMean = pd.DataFrame(plotMean).T
-
-    plotMean[['MLIR TFlops', 'MIOpen TFlops']]\
-        .to_csv(reportUtils.PERF_PLOT_REPORT_FILE, index=False)
-
-    means = df.groupby(["Direction", "DataType"])[COLUMNS_TO_AVERAGE]\
-        .agg(reportUtils.geoMean)
-    means.loc[("All", "All"),:] = df[COLUMNS_TO_AVERAGE].agg(reportUtils.geoMean)
-    means.to_csv(reportUtils.PERF_STATS_REPORT_FILE)
-
-    with open("MLIR_vs_MIOpen.html", 'w') as htmlOutput:
-        reportUtils.htmlReport(df, means, "MLIR vs. MIOpen performance", "MLIR/MIOpen", htmlOutput)
-
-def generatePerformanceResults(configs, xdlops):
-    mlir_df = pd.DataFrame(benchmarkMLIR(testVector.split(sep=' '), xdlops) for testVector in configs)
-    miopen_df = pd.DataFrame(benchmarkMIOpen(testVector.split(sep=' '), xdlops) for testVector in configs)
-
-    printPerformance(mlir_df, miopen_df)
+    with pd.option_context('precision', reportUtils.ROUND_DIGITS):
+        print(df)
+
+def getSolverName(testVector, xdlops):
+    config = ConvConfiguration.fromCommandLine(testVector.split(sep=' '), xdlops)
+    if config.direction == 'fwd':
+       solverName = 'ConvMlirIgemmFwd'
+    elif config.direction == 'bwd':
+       solverName = 'ConvMlirIgemmBwd'
+    else:
+       solverName = 'ConvMlirIgemmWrW'
+    if xdlops == True:
+       solverName+='Xdlops'
+    return solverName
+    
+def benchmarkMIOpenWithMLIRKernels(configs, xdlops):
+    solver_names = {testVector : getSolverName(testVector, xdlops) for testVector in configs}
+
+    # Set environment variables for running MIOpenDriver with MLIR kernels
+    envs = dict(MIOPEN_FIND_MODE='1', MIOPEN_DRIVER_USE_GPU_REFERENCE='l')
+    perf_list = []
+    for testVector in configs:
+        envs['MIOPEN_DEBUG_FIND_ONLY_SOLVER']=solver_names[testVector]
+        perf_list.append(benchmarkMIOpen(testVector.split(sep=' '), xdlops, envs))
+    df = pd.DataFrame(perf_list)
+    df.to_csv(reportUtils.MIOPEN_REPORT_FILE, index=False)
+
+#Tune MIOpen with MLIR kernels
+def tuneMLIRKernels(configs, xdlops):
+    solver_names = {testVector : getSolverName(testVector, xdlops) for testVector in configs}
+
+    envs = dict(MIOPEN_FIND_ENFORCE='4', MIOPEN_DRIVER_USE_GPU_REFERENCE='1')
+    for testVector in configs:
+        envs['MIOPEN_DEBUG_FIND_ONLY_SOLVER']=solver_names[testVector]
+        commandLine = testVector.split(sep=' ')
+        config = ConvConfiguration.fromCommandLine(commandLine, xdlops)
+        if config.inputLayout == 'nchw':
+            MIOpenDriverCommand = MIOPEN_DRIVER + ' ' + ' '.join(commandLine) + ' -V 0'
+            print(MIOpenDriverCommand)
+            p1 = subprocess.Popen(MIOpenDriverCommand.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=envs)
+            # get output.
+            try:
+               outs, errs = p1.communicate(timeout=180)
```

Comment:
Since we're getting the output, can we slide a check in here to make sure we've output what's expected?

Reply:
The validation tests have the check to make sure MIOpen ran correctly - maybe we can adapt that?

Reply:
Performance tests purposely turn off validation ('-V 0') to make more reasonable time results. I think we can assume MIOpen runs correctly here, as we also run validation tests separately.

Reply:
My main concern here is making sure we don't accidentally compare against a no-op "run", but that can be a seperate PR.

Reply:
> Minor nitpick about `unstash` aside, looks good to me
> 
> I'm assuming you've run these tests by hand to make sure the report generation is successful?

Yes. The tests just finished.
http://ml-ci.amd.com:21096/job/MLIR/job/MLIR-tuned-kernel-perf/